### PR TITLE
Add sphinx to mypy deps in pre-commit and test-requirements

### PR DESCRIPTION
**Pull request by cidrblock**
Ongoing effort to remove ignore-imports from mypy.

- Added as a dependency in the precommit entries for python3.6 and 3.10
- Added to test-requirements so mypy can be used from the command line
- One type ignore b/c sphinx spelling is not typed.

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@ flake8-docstrings
 flake8-quotes
 libtmux
 lxml
+sphinx
```
Comment (webknjaz) [inline test-requirements.txt:7]: Why put it here? It's not related to tests...

Comment (cidrblock) [reply test-requirements.txt:7]: should we have a dev-requirements file too?

Comment (cidrblock) [reply test-requirements.txt:7]: Mypy does need it, and mypy could be considered a "test"....?

**Review by relrod — APPROVED** [review]

File: test-requirements.txt (hunk 1 above)
Comment (cidrblock) [reply test-requirements.txt]: @webknjaz let's iterate on where these should be. Currently test-requirements represents everything a dev would, or soon will.

**Pull request merged**